packages/react-ui-core/src/Modal/Modal.js

```diff
@@ -93,6 +99,12 @@ export default class Modal extends PureComponent {
     if (this.props.onClose) this.props.onClose()
   }
 
+  toggleBodyClass = toggle => {
```

Comment:
I moved this responsibility to the Modal component so that Modals themselves can decide whether or not the body should have the `-lock` classname applied to it.